WordPress/Classes/Services/BlogService.m

```diff
@@ -159,9 +159,14 @@ - (void)syncBlogsForAccount:(WPAccount *)account
 {
     DDLogMethod();
 
+    NSManagedObjectID *accountID = account.objectID;
     id<AccountServiceRemote> remote = [self remoteForAccount:account];
     [remote getBlogsWithSuccess:^(NSArray *blogs) {
         [self.managedObjectContext performBlock:^{
+            WPAccount *account = (WPAccount *)[self.managedObjectContext existingObjectWithID:accountID error:nil];
+            if (!account) {
+                return;
```

Comment:
I see two possible solutions here:

 * Always use NSManageObjectID for the parameters, because we are unsure what thread/context they are going to be use.
 * Always assume that the calls to services need to be done on the main context, then internally we decide if we need to get the object ID or not depending of what kind of context we are using.

We just need to be consistent about it or else is very hard to reason about the code.